### PR TITLE
Fieldsets / Wells

### PR DESCRIPTION
**Pull request by jasonfellows**
I don't think we need to pass props for 'field-light', et al. to all of our field components. This can be accomplished by styling children of the native `fieldset` node and the `.tooltip` class.

I'd started working on a FieldSet component and it was completely dumb, no props or any logic, just a wrapper adding functional classes.

There are a few things like PTO plan settings that use a well-style appearance, but these can be styled using functional classes. Default styles should cover all other cases.

screen shot 2015-12-21 at 5 32 35 pm
screen shot 2015-12-21 at 5 32 41 pm

This PR covers the following tasks:
- [x] Update forms.js to use native `fieldset`
- [x] Add default style for `fieldset` element
- [x] Update `forms.scss` to style children of `fieldset`
- [x] Create styles in `forms.scss` for forms inside popups
- [x] Fix missing / incorrect hover styles for `inputs` in all three parents 
- [x] Remove `fieldColor` props from all fields

### CODE REVIEW

**Comment by keithpops** [comment]
I don't have anything handy to reference at the moment, but don't some of the horizontal rules go end to end without padding? If we do, is this something we would want to add into this or should we keep those as one-offs for HCM?

**Comment by jasonfellows** [comment]
We talked about this a little bit. Let's treat those as one-offs or use border on the container object. I think the example was time off plans?

**Comment by keithpops** [comment]
Ah that's right. I'm fine with that.

File: src/scss/elements/_forms.scss [hunk 1]
```diff
@@ -1,63 +1,71 @@
...
+    color: transparent;
+  }
+
   &:focus {
     outline: 0;
   }
 
   &.bc-orange {
-    border-color: $orange;
+    border-color: $orange !important;
```
Comment (keithpops) [inline src/scss/elements/_forms.scss]: ruh roh - what's the `important!` for?

Comment (jasonfellows) [inline src/scss/elements/_forms.scss]: Artificial specificity.

**Comment by jasonfellows** [comment]
Somebody receiving emails from this PR left their iPhone 6 in the men's restroom. Please go to the front desk to pick it up.

File: src/js/components/forms/fields/checkbox.es6 [hunk 2]
```diff
@@ -56,15 +48,11 @@ export default React.createClass({
...
-          <input
-            checked={this.state.isChecked}
-            className={this.fieldClasses()}
-            disabled={this.props.disabled}
-            onChange={this.onChange}
-            readOnly={this.props.readOnly}
-            type="checkbox"
-          >
-          </input>
+          <input checked={this.state.isChecked}
```
Comment (HipsterBrown) [inline src/js/components/forms/fields/checkbox.es6]: Would this ever be checked by default? If so, then it might need to be passed in as a prop.

Comment (HipsterBrown) [inline src/js/components/forms/fields/checkbox.es6]: Oops nevermind, I didn't see the props because of code folding. :+1:

Comment (jasonfellows) [inline src/js/components/forms/fields/checkbox.es6]: Thats untouched by this PR, but there is already a prop called `checked` that sets the initial state. I would suggest renaming this to `initialValue`, which is a part of the common field interface.

File: src/scss/elements/_forms.scss [hunk 3]
```diff
@@ -1,63 +1,71 @@
 input,
 select,
-textarea,
-.simple-select,
-fieldset {
+textarea {
+  -webkit-appearance: none;
```
Comment (HipsterBrown) [inline src/scss/elements/_forms.scss]: Isn't `appearance` prefixes included with Autoprefixer? https://github.com/postcss/autoprefixer/issues/43

Comment (jasonfellows) [inline src/scss/elements/_forms.scss]: Good point. This is also unchanged by this PR.

Comment (HipsterBrown) [inline src/scss/elements/_forms.scss]: Ah I mistook the diff.

**Comment by HipsterBrown** [comment]
Looks good to me. :+1: 

I really enjoy that dark field UI. A great color scheme.

**Comment by katienreed** [comment]
+1 looks good to me, Jason!

**Pull request merged**